### PR TITLE
Move table `addons` into `node.config.php`

### PR DESCRIPTION
**Pull request by nupplaphil**
So this is the next step :-)

- Moved only the relevant addon columns into a new config category `addons`, I skipped:
  - `version` is always empty and never used
  - `installed` is always `true`
  - `hidden` is always `false` because there's no `.hidden` file at any addon
- Migrated the whole table to the `node.config.php`
- Replaced the logic at every place I know

Still I found a general bug in the new process, which I do want to discuss:
There **seems** to be a delay when using `file_put_contents()`.

- When I activate an addon, the addon is correctly saved into the `node.config.php`.
- There's a `$this->baseurl->redirect('admin/addons');` after the install process
- The success message appears, but the addon isn't checked (same behavior for deactivate)
- But if I manually reload the page or press the "reload active addons" button, the correct state returns

Additionally, if I remove the redirect, the addon is shown correctly (but the whole `toggle` url-query appears, which is the reason for the redirect).

I didn't find any references about such behavior, so I struggle a little bit tbh

### CODE REVIEW

**Comment by MrPetovan** [comment]
Could it be the browser cached version of the page that gets refreshed when you force reload?

**Comment by nupplaphil** [comment]
After merging #12606 , this is ready as well :)
Both are running at my friendica.philipp.info

I will search for any errors tomorrow

File: src/Core/Addon.php [hunk 1]
```diff
@@ -168,12 +169,9 @@ public static function install(string $addon): bool
 			$func(DI::app());
 		}
 
-		DBA::insert('addon', [
-			'name' => $addon,
-			'installed' => true,
-			'timestamp' => $t,
-			'plugin_admin' => function_exists($addon . '_addon_admin'),
-			'hidden' => file_exists('addon/' . $addon . '/.hidden')
+		DI::config()->set('addons', $addon,[
```
Comment (MrPetovan) [inline src/Core/Addon.php]:
```suggestion
		DI::config()->set('addons', $addon, [
```

File: static/settings.config.php [hunk 2]
```diff
@@ -52,10 +52,6 @@
 		// Enter 0 for no time limit.
 		'account_abandon_days' => 0,
 
-		// addon (Comma-separated list)
-		// Manual list of addons which are enabled on this system.
-		'addon' => '',
```
Comment (MrPetovan) [inline static/settings.config.php:57]: Don't forget to remove the mentions to this configuration key in the various addon install documentation.

**Review by MrPetovan — CHANGES_REQUESTED** [review]

**Review by MrPetovan — APPROVED** [review]

File: static/settings.config.php (hunk 2 above)
Comment (nupplaphil) [reply static/settings.config.php:57]: thx: https://github.com/friendica/friendica-addons/pull/1345

File: update.php [hunk 3]
```diff
@@ -1192,5 +1192,23 @@ function update_1508()
...
+	foreach ($addons as $addon) {
+		$newConfig->set('addons', $addon['name'], [
+			'last_update' => $addon['timestamp'],
+			'admin' => (bool)$addon['plugin_admin'],
+		]);
+	}
+
+	$newConfig->commit();
+
+	return DBA::e("TRUNCATE TABLE `addon`") ? Update::SUCCESS : Update::FAILED;
```
Comment (annando) [inline update.php:1213]: Why are you deleting the data? In case of some update problem this will result in data loss.

Comment (MrPetovan) [reply update.php:1213]: The config writer will throw an Exception if there’s an update problem, which will prevent the deletion.

Comment (annando) [reply update.php:1213]: We don't gain any significant advantage from deleting the data. In the past we simply added the table to the list of the tables that can be deleted manually when one feels safe about it.

Edit: How do you want to trace some possible "some parts of my configuration are lost" support requests? When we can't compare the outcome with the source, we don't have got any possibility to track down possible problems.

Comment (MrPetovan) [reply update.php:1213]: I suppose this is a good safety measures when a system can *silently* break in *numerous unexpected ways* as it was historically the case in Friendica. We are currently trying to reduce this liability with the use of dependency injection and tests and exceptions and structured data which makes this strategy less and less useful over time.

Still, given the relative small size of the config and addon tables, I don't think it's a big deal if we leave the data.

**Pull request merged**